src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -28,10 +28,18 @@ public static RecordValue Empty()
             var type = new RecordType();
             return new InMemoryRecordValue(IRContext.NotInSource(type), new List<NamedValue>());
         }
+         
+        /// <summary>
+        /// Get a field on this record. 
+        /// See <see cref="ObjectMarshaler"/> to override GetField behavior. 
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public FormulaValue GetField(string name)
+        {            
+            var fieldType = ((RecordType)Type).MaybeGetFieldType(name) ?? FormulaType.Blank;
 
-        public virtual FormulaValue GetField(string name)
-        {
-            return GetField(IRContext.NotInSource(FormulaType.Blank), name);
```

Comment:
>FormulaType.Blank) [](http://example.com/codeflow?start=50&length=18)

existing bug caught by new tests. Blank is wrong when missing a field in a heterogenous table.